src/components/UserProfile/UserProfileAccount/index.js

```diff
@@ -1,53 +1,78 @@
+import { LoaderContent } from "../../../utility/constants/LoaderContent";
 
-const walletDropdownList = Object.entries(walletTypes).map(([key, value]) => ({ value, label: key }));
-
-const UserProfileAccount = ({ classes, wallet, updateWallet, stopWalletDetailsPolling }) => {
+const UserProfileAccount = ({ classes, dispatch }) => {
   const [alert, setAlert] = useState({});
+  const initialWallet = { value: "default", type: "default" };
+  const [wallet, updateWallet] = useState(initialWallet);
+  const [wallets, updateWallets] = useState([]);
+  const [linkedProviders, updateLinkedProviders] = useState([]);
+  useEffect(() => {
+    const fetchWallets = async () => {
+      const availableWallets = await fetchAvailableUserWallets();
+      const enhancedWallets = map(availableWallets, ({ address, type }) => {
+        return { address, type, value: address, label: `${type} (${address})` };
+      });
+      updateWallets(enhancedWallets);
+    };
+    fetchWallets();
+  }, []);
 
+  const selectedEthAddress = window.ethereum && window.ethereum.selectedAddress;
+  const isSameMetaMaskAddress = lowerCase(selectedEthAddress) === lowerCase(wallet.address);
   const handleWalletTypeChange = async event => {
     setAlert({});
-    const { value } = event.target;
-    if (value === walletTypes.METAMASK) {
+    const { value: selectedValue } = event.target;
+    const selectedWallet = find(wallets, ({ value }) => selectedValue === value);
+    if (!selectedWallet) {
+      updateWallet(initialWallet);
+      return;
+    }
+
+    dispatch(startAppLoader(LoaderContent.FETCH_LINKED_PROVIDERS));
+    const organizations = await fetchWalletLinkedProviders(selectedWallet.address);
+    dispatch(stopAppLoader);
+    updateLinkedProviders(organizations);
+    updateWallet(selectedWallet);
+
+    if (selectedWallet.type === walletTypes.METAMASK) {
       try {
-        const selectedEthAddress = window.ethereum && window.ethereum.selectedAddress;
-        const sdk = await initSdk(selectedEthAddress);
-        const address = sdk.account.address;
-        if (!isEmpty(address)) {
-          stopWalletDetailsPolling();
-          updateWallet({ type: value, address });
+        if (isSameMetaMaskAddress) {
+          await initSdk(selectedEthAddress);
           return;
         }
         setAlert({ type: alertTypes.ERROR, message: `Unable to fetch Metamask address. Please try again` });
       } catch (error) {
         setAlert({ type: alertTypes.ERROR, message: `Something went wrong. Please try again` });
       }
     }
-    if (value === walletTypes.GENERAL) {
-      stopWalletDetailsPolling();
-      updateWallet({ type: value });
-      return;
-    }
-
-    updateWallet({ type: value });
   };
 
   const walletDetails = {
-    [walletTypes.METAMASK]: <MetamaskDetails />,
+    [walletTypes.METAMASK]: isSameMetaMaskAddress ? <MetamaskDetails wallet={wallet} /> : undefined,
```

Comment:
should we have to render some Error text instead of undefined if it is not same MM address